apps/trading/components/rewards-container/streaks/activity-streaks.tsx

```diff
@@ -0,0 +1,218 @@
+import { VegaIcon, VegaIconNames } from '@vegaprotocol/ui-toolkit';
+import { useT } from '../../../lib/use-t';
+import classNames from 'classnames';
+import { formatNumber } from '@vegaprotocol/utils';
+import BigNumber from 'bignumber.js';
+
+export const ActivityStreak = ({
+  tiers,
+  streak,
+}: {
+  tiers:
+    | never[]
+    | {
+        minimum_activity_streak: number;
+        reward_multiplier: string;
+        vesting_multiplier: string;
+      }[];
+  streak:
+    | (
+        | {
+            activeFor: number;
+            isActive: boolean;
+            inactiveFor: number;
+            rewardDistributionMultiplier: string;
+            rewardVestingMultiplier: string;
+            epoch: number;
+            tradedVolume: string;
+            openVolume: string;
+          }
+        | null
+        | undefined
+      )
+    | undefined;
+}) => {
+  const t = useT();
+
+  const getUserTier = () => {
+    if (!streak?.activeFor) return 0;
+    let userTier = 0,
+      i = 0;
+    while (
+      i < tiers.length &&
+      tiers[userTier].minimum_activity_streak < streak.activeFor
+    ) {
+      userTier = i;
+      i++;
+    }
+
+    if (
+      i === tiers.length &&
+      tiers[userTier].minimum_activity_streak <= streak.activeFor
+    ) {
+      userTier = i;
+    }
+
+    if (userTier > tiers.length) {
+      userTier--;
+    }
+
+    return userTier;
+  };
+
+  if (!tiers || tiers.length === 0) return null;
+
+  const userTierIndex = getUserTier();
+
+  const safeProgress = (i: number) => {
+    if (i < userTierIndex) return 100;
+    if (i > userTierIndex) return 0;
+    if (!streak?.activeFor) return 0;
+    const progress = streak?.activeFor;
+    const total = tiers[i].minimum_activity_streak;
+    if (!total) return 0;
+    if (new BigNumber(progress).isGreaterThan(total)) return 100;
+    return new BigNumber(progress)
+      .multipliedBy(100)
+      .dividedBy(total || 1)
+      .toNumber();
+  };
+  const progressBarHeight = 'h-10';
+
+  return (
+    <>
+      <div className="flex flex-col gap-1 w-full">
+        <div className="flex flex-col gap-1">
+          <div
+            className="grid"
+            style={{
+              gridTemplateColumns:
```

Comment:
I'm not sure but would flex or auto columns not work here?

Reply:
the streaks could have 12 tiers, after 4 tiers they should have another grid template.

Reply:
I am not using flex for this 
<img width="805" alt="image" src="https://github.com/vegaprotocol/frontend-monorepo/assets/16125548/073743a2-c436-4f2f-ae72-5552240dacd2">